kolena/_experimental/dataset/common.py

```diff
@@ -11,14 +11,41 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+from collections import Counter
+from typing import List
+
+import pandas as pd
+
+from kolena._utils import log
 from kolena.errors import InputValidationError
 
 
+COL_DATAPOINT = "datapoint"
+COL_DATAPOINT_ID_OBJECT = "datapoint_id_object"
+COL_EVAL_CONFIG = "eval_config"
+COL_RESULT = "result"
+
+
 def validate_batch_size(batch_size: int) -> None:
     if batch_size <= 0:
         raise InputValidationError(f"invalid batch_size '{batch_size}': expected positive integer")
 
 
-COL_DATAPOINT = "datapoint"
-COL_EVAL_CONFIG = "eval_config"
-COL_RESULT = "result"
+def validate_id_fields(df: pd.DataFrame, id_fields: List[str], existing_id_fields: List[str] = None) -> None:
+    if len(id_fields) == 0:
+        raise InputValidationError(f"invalid id_fields '{id_fields}': expected at least one field")
+    if existing_id_fields:
+        if Counter(id_fields) != Counter(existing_id_fields):
```

Comment:
should we use `set` at here?

Reply:
set does not validate on the dups, but we won't have dups?

Reply:
oh, i see. maybe we need to do an explicit no dups in `id_fields`? b/c the current warning msg is only for the changes

Reply:
got it I will add explicit no dup check